### PR TITLE
check cron expression valid

### PR DESCRIPTION
**Pull request by qingyuanz**
If it is parsing some wrong expression, like (*/2 * * 10-5 *), when calling Next(), it panics in some cases.
So should check the cron expression valid first.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (irvinlim) [inline .gitignore]: nit: I think we don't need to add .gitignore to the fork, we can try to minimize changes to upstream.

**Review by irvinlim — APPROVED** [review]
@qingyuanz Thanks for your PR 😄  we should also probably update the version in the https://github.com/furiko-io/furiko project as well!

File: .gitignore (hunk 1 above)
Comment (qingyuanz) [reply .gitignore]: done

**Comment by irvinlim** [comment]
LGTM! 🙇

**Comment by irvinlim** [comment]
@qingyuanz Tagged as v0.1.3 (in order to maintain parity with internal tags). We could proceed to update the main furiko code now!

**Pull request merged**